core/src/main/java/com/es/core/dao/phone/JdbcPhoneDao.java

```diff
@@ -0,0 +1,100 @@
+                "batteryCapacityMah, talkTimeHours, standByTimeHours, bluetooth, positioning, imageUrl, " +
+                "description) " +
+                "VALUES (:brand, :model, :price, :displaySizeInches, :weightGr, :lengthMm, " +
+                ":widthMm, :heightMm, :announced, :deviceType, :os, :displayResolution, :pixelDensity, " +
+                ":displayTechnology, :backCameraMegapixels, :frontCameraMegapixels, :ramGb, :internalStorageGb, " +
+                ":batteryCapacityMah, :talkTimeHours, :standByTimeHours, :bluetooth, :positioning, :imageUrl, " +
+                ":description)";
+
+        SqlParameterSource namedParameters = new BeanPropertySqlParameterSource(phone);
+        namedParameterJdbcTemplate.update(SQL, namedParameters, keyHolder, new String[]{"id"});
+        return (Long) keyHolder.getKey();
+    }
+
+    public List<Phone> findAll(int offset, int limit) {
+       String SQL = "SELECT * FROM phones LEFT JOIN phone2color ON phones.id=phone2color.phoneId " +
+                "LEFT JOIN colors ON phone2color.colorId = colors.id " +
+                "WHERE phones.price > 0 AND " +
+                "((SELECT stocks.stock FROM stocks WHERE stocks.phoneId = phones.id) > 0 )" +
+            "OFFSET " + offset + " LIMIT " + limit;
+       try {
+           List<Phone> phones = (List<Phone>) jdbcTemplate.query(SQL, new PhoneResultSetExtractor());
+           return phones;
+       } catch (EmptyResultDataAccessException e) {
+           return new ArrayList<>();
+       }
+    }
+
+    @Override
+    public List<Phone> sortByField(SortField sortField, SortType sortType, int limit) {
+        String SQL = "SELECT * FROM phones LEFT JOIN phone2color ON phones.id=phone2color.phoneId " +
+                "LEFT JOIN colors ON phone2color.colorId = colors.id " +
+                "WHERE phones.price > 0 AND " +
+                "((SELECT stocks.stock FROM stocks WHERE stocks.phoneId = phones.id) > 0 )" +
+                "ORDER BY UPPER( " +sortField.field + " ) " + sortType.type + " LIMIT " + limit;
+        try {
+            List<Phone> phones = (List<Phone>) jdbcTemplate.query(SQL, new PhoneResultSetExtractor());
+            return phones;
+        } catch (EmptyResultDataAccessException e) {
+            return new ArrayList<>();
+        }
+    }
+
+    @Override
+    public List<Phone> userSearchByModel(String searchString, int limit) {
+        String SQL = "SELECT * FROM phones LEFT JOIN phone2color ON phones.id=phone2color.phoneId " +
+                "LEFT JOIN colors ON phone2color.colorId = colors.id " +
+                "WHERE phones.price > 0 AND " +
+                "((SELECT stocks.stock FROM stocks WHERE stocks.phoneId = phones.id) > 0 ) " +
+                "AND model LIKE '%" + searchString.trim() + "%'";
+        try {
+            List<Phone> phones = (List<Phone>) jdbcTemplate.query(SQL, new PhoneResultSetExtractor());
```

Comment:
That's a bad idea to use `ResultSetExtractor` for both findOne() and findAll(). Use either `RowCallbackHandler` instead or `BeanPropertyRowMapper`. As I said, it's ok not to extract additional data for non-single result methods.